examples/rdl.py

```diff
@@ -0,0 +1,773 @@
+        )
+
+        # Create a dictionary mapping edge types to number of neighbors
+        num_neighbors_dict = {
+            edge_type: args.num_neighbors  # [num_neighbors for each direction]
+            for edge_type in data.edge_types
+        }
+
+        loader_dict[split] = NeighborLoader(
+            data=data,
+            num_neighbors=num_neighbors_dict,
+            input_nodes=table_input.nodes,
+            input_time=table_input.time,
+            time_attr="time",
+            transform=table_input.transform,
+            batch_size=args.batch_size,
+            temporal_strategy=args.temporal_strategy,
+            shuffle=split == "train",
+            num_workers=1,
+            persistent_workers=True,
+        )
+
+    # Get task-specific parameters
+    print("Getting task-specific parameters...")
+    out_channels, loss_fn, tune_metric, higher_is_better = \
+        get_task_type_params(task)
+    print("out_channels: ", out_channels)
+    print("loss_fn: ", loss_fn)
+    print("tune_metric: ", tune_metric)
+    print("higher_is_better: ", higher_is_better)
+
+    # Define the model
+    print("Initializing the model...")
+    node_types = data.node_types  # Include all node types
+    edge_types = data.edge_types  # Include all edge types
+    col_names_dict = {
+        node_type: data[node_type].tf.col_names_dict
+        for node_type in data.node_types
+    }
+    temporal_node_types = [
+        node_type for node_type in data.node_types if "time" in data[node_type]
+    ]
+    model = Model(
+        node_types=node_types,
+        edge_types=edge_types,
+        col_names_dict=col_names_dict,
+        col_stats_dict=col_stats_dict,
+        temporal_node_types=temporal_node_types,
+        num_layers=len(args.num_neighbors),
+        channels=args.channels,
+        out_channels=out_channels,
+        aggr=args.aggr,
+        norm=args.norm,
+    ).to(device)
+
+    # Define the optimizer
+    optimizer = torch.optim.Adam(model.parameters(), lr=args.lr)
+
+    print("Training the model...")
+    state_dict = None
+    best_val_metric = -math.inf if higher_is_better else math.inf
+    for epoch in range(1, args.epochs + 1):
+        train_loss = train(
+            model=model,
+            train_loader=loader_dict["train"],
+            task=task,
+            optimizer=optimizer,
+            loss_fn=loss_fn,
+        )
+        val_pred = test(
+            test_loader=loader_dict["val"],
+            model=model,
+            task=task,
+        )
+        val_metrics = task.evaluate(val_pred, val_table)
+        print(f"Epoch: {epoch:02d}, "
+              f"Train loss: {train_loss}, "
+              f"Val metrics: {val_metrics}")
+
+        is_better_op = operator.gt if higher_is_better else operator.lt
+        if is_better_op(val_metrics[tune_metric], best_val_metric):
+            best_val_metric = val_metrics[tune_metric]
+            state_dict = copy.deepcopy(model.state_dict())
```

Comment:
It doesn't matter much for this model size, but it'd be good to offload the state dict to CPU :)

Reply:
ah, good point - feel free to add :)